pypeit/flatfield.py

```diff
@@ -110,21 +130,59 @@ def _bundle(self):
             if self[key] is None:
                 continue
             # Array?
-            if self.datamodel[key]['otype'] == np.ndarray and key != 'spat_bsplines':
+            if self.datamodel[key]['otype'] == np.ndarray and \
+                    key != 'pixelflat_spat_bsplines' and \
+                    key != 'illumflat_spat_bsplines':
                 d.append({key: self[key]})
-            elif key == 'spat_bsplines':
+            elif key == 'pixelflat_spat_bsplines':
                 for ss, spat_bspl in enumerate(self[key]):
                     # Naming
-                    spat_bspl.hdu_prefix = 'SPAT_ID-{}_'.format(self.spat_id[ss])
+                    spat_bspl.hdu_prefix = 'PIXELFLAT_SPAT_ID-{}_'.format(self.spat_id[ss])
                     dkey = 'bspline-{}'.format(self.spat_id[ss])
                     # Save
-                    d.append({dkey: spat_bspl})
+                    d.append(copy.deepcopy({dkey: spat_bspl}))
+            elif key == 'illumflat_spat_bsplines':
+                for ss, spat_bspl in enumerate(self[key]):
+                    # Naming
+                    spat_bspl.hdu_prefix = 'ILLUMFLAT_SPAT_ID-{}_'.format(self.spat_id[ss])
+                    dkey = 'bspline-{}'.format(self.spat_id[ss])
+                    # Save
+                    d.append(copy.deepcopy({dkey: spat_bspl}))
             else: # Add to header of the primary image
                 d[0][key] = self[key]
         # Return
         return d
 
```

Comment:
I am not sure I'm following while these have been converted from attributes to properties? Can you elaborate on the logic behind that change?